packages/storybook-block/src/StorybookBlock.tsx

```diff
@@ -47,58 +44,40 @@ export const StorybookBlock: FC<BlockProps> = ({ appBridge }) => {
         radiusValue = '',
     } = blockSettings;
 
-    const deleteUrl = () => {
-        setIframeUrl(null);
-        setLocalUrl('');
-        setBlockSettings({
-            ...blockSettings,
-            url: '',
-        });
-    };
+    const isEditing = useEditorState(appBridge);
+    const [input, setInput] = useState(url);
+    const [storybookUrl, setStorybookUrl] = useState(url);
+    const { hoverProps, isHovered } = useHover({});
+    const { containerRef, setIsReadyForPrint } = useReadyForPrint();
 
-    const saveLink = () => {
+    const iframeUrl = buildIframeUrl(storybookUrl, style === StorybookStyle.Default, positioning);
+    const saveInputLink = useCallback(() => {
         setBlockSettings({
             ...blockSettings,
-            url: localUrl,
+            url: ensureHttps(input),
         });
-    };
+    }, [blockSettings, input, setBlockSettings]);
 
     useEffect(() => {
-        if (url !== '') {
-            setIsReadyForPrint(false);
-            const newIframeUrl = new URL(url);
-            newIframeUrl.searchParams.set('nav', 'false');
-
-            const hasAddons = style === StorybookStyle.Default;
-            const shouldAddIframeToUrl = !hasAddons;
-            const includesIframe = newIframeUrl.pathname.toString().includes('iframe.html');
-            const positionValue = positioning === StorybookPosition.Horizontal ? 'right' : 'bottom';
-            const panelValue = !hasAddons ? 'false' : positionValue;
-
-            newIframeUrl.searchParams.set('panel', panelValue);
-
-            if (shouldAddIframeToUrl && !includesIframe) {
-                newIframeUrl.pathname = `${newIframeUrl.pathname}iframe.html`;
-            }
-
-            if (!shouldAddIframeToUrl && includesIframe) {
-                const pathname = newIframeUrl.pathname.toString().replace('iframe.html', '');
-                newIframeUrl.pathname = pathname;
-            }
-
-            setIframeUrl(newIframeUrl);
-        } else if (url === '') {
-            setIsReadyForPrint(true);
-            deleteUrl();
-        }
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [url, style, positioning]);
+        setStorybookUrl(url);
+        setIsReadyForPrint(url !== '');
```

Comment:
Block should initially be ready for print.
As soon as a link is entered it should NOT be ready.
And as soon as the iframe is loaded (or the URL is invalid) it should be ready again.

Currently the block is NOT ready when no URL is entered.